### PR TITLE
feat: [say.c] Added language support for pashto and dari languages 

### PR DESCRIPTION
**Pull request by iamtalhaasghar**
This pull request to add language support for Pashto and Dari in the mod_say module in asterisk is a great addition to the project. With this new feature, users who speak these languages can now benefit from the text-to-speech functionality provided by asterisk. This will make the platform more accessible and useful to a wider range of users, particularly those in regions where Pashto and Dari are spoken. This contribution will help to improve the overall usability and inclusivity of the asterisk platform.

### CODE REVIEW

File: main/say.c [hunk 1]
```diff
@@ -545,7 +545,7 @@ struct ast_str* ast_get_number_str(int num, const char *lang)
 {
 	if (!strncasecmp(lang, "en", 2)) { /* English syntax */
 		return get_number_str_en(num, lang);
-	}
+    }
```
Comment (InterLinked1) [inline main/say.c:548]: You've messed up the whitespace here

File: main/say.c [hunk 2]
```diff
@@ -900,9 +904,12 @@ static int say_number_full(struct ast_channel *chan, int num, const char *ints,
 		return ast_say_number_full_ur(chan, num, ints, language, options, audiofd, ctrlfd);
 	} else if (!strncasecmp(language, "vi", 2)) { /* Vietnamese syntax */
 		return ast_say_number_full_vi(chan, num, ints, language, audiofd, ctrlfd);
-	}
-
-	/* Default to english */
+	} else if (!strncasecmp(language, "dr", 2)) { /* Dari syntax */
+		return ast_say_number_full_dr(chan, num, ints, language, audiofd, ctrlfd);
+    } else if (!strncasecmp(language, "ps", 2)) { /* Pashto syntax */
```
Comment (InterLinked1) [inline main/say.c:909]: Whitespace messed up

File: main/say.c [hunk 3]
```diff
@@ -1293,6 +1300,158 @@ static int ast_say_number_full_de(struct ast_channel *chan, int num, const char
...
+	int mf = 1;                            /* +1 = male and neuter; -1 = female */
+	char fn[256] = "";
+	char fna[256] = "";
+	if (!num)
+		return ast_say_digits_full(chan, 0, ints, language, audiofd, ctrlfd);
+
+	if (options && (!strncasecmp(options, "f", 1)))
+		mf = -1;
+
+    int playWesht = 0;
```
Comment (InterLinked1) [inline main/say.c:1323]: Also messed up here. And you can't declare variables after use, compile with developer mode and your code will probably fail to build.

You need to fix all the whitespace and variable misuse in this change.

**Review by InterLinked1 — CHANGES_REQUESTED** [review]
- CLA is not signed yet
- PR needs to be squashed into a single commit
- Commit message does not comply with coding guidelines
- Lots of coding guideline violations in the change that need to be fixed

File: main/say.c (hunk 3 above)
Comment (iamtalhaasghar) [reply main/say.c:1323]: oh, crap looks like my vim settings messed up white spaces / tabs.

**Review by gtjoseph — CHANGES_REQUESTED** [review]
Please address all of @InterLinked1's comments.

**Comment by iamtalhaasghar** [comment]
Yeah, i will review and do the required changes on coming weekend!

**Review by iamtalhaasghar — COMMENTED** [review]
Done requested changes!

**Comment by iamtalhaasghar** [comment]
- [x]  CLA is not signed yet
- [x] PR needs to be squashed into a single commit
- [x] Commit message does not comply with coding guidelines
- [x] Lots of coding guideline violations in the change that need to be fixed

@gtjoseph @InterLinked1 Pls review!

**Review by jcolp — APPROVED** [review]

**Comment by iamtalhaasghar** [comment]
Can you kindly merge this PR? @jcolp

**Comment by jcolp** [comment]
PRs require two people to review them before merging. That has not happened as of yet.

File: main/say.c [hunk 4]
```diff
@@ -1293,6 +1301,158 @@ static int ast_say_number_full_de(struct ast_channel *chan, int num, const char
...
+				}
+				num = 0;
+			}
+		} else if (num == 100 && t == 0) {
+			ast_copy_string(fn, "digits/hundred", sizeof(fn));
+			num = 0;
+		} else if (num < 1000) {
+			int hundreds = num / 100;
+			num = num % 100;
+			// 100: hundreds = 1, num = 0
```
Comment (InterLinked1) [inline main/say.c:1371]: // C99 comments are not allowed. Use regular comments.

**Review by InterLinked1 — CHANGES_REQUESTED** [review]

**Comment by asteriskteam** [comment]
This PR has been marked stale because it has been in "Changes Requested" or "submitter-action-required" state for 28 days or more. Please make the requested changes within 14 days or the PR will be closed.

**Review by gtjoseph — CHANGES_REQUESTED** [review]
There are still issues in this PR

- [ ] Address @InterLinked1's comments 
- [ ] There aren't any "cherry-pick" comments.
- [ ] There should be an "Improvement" issue opened and both the commit message and the PR description should reference it.

See:
https://docs.asterisk.org/Development/Policies-and-Procedures/Code-Contribution/
https://docs.asterisk.org/Development/Policies-and-Procedures/Commit-Messages/

**Comment by asteriskteam** [comment]
This PR has been marked stale because it has been in "Changes Requested" or "submitter-action-required" state for 28 days or more. Please make the requested changes within 14 days or the PR will be closed.

**Comment by gtjoseph** [comment]
You've got a merge commit now which must be removed.  You'll need to rebase your branch off the current master and re-force-push.

**Comment by iamtalhaasghar** [comment]
> You've got a merge commit now which must be removed. You'll need to rebase your branch off the current master and re-force-push.

Ok i will clean it up.

**Comment by asteriskteam** [comment]
This PR has been marked stale because it has been in "Changes Requested" or "submitter-action-required" state for 28 days or more. Please make the requested changes within 14 days or the PR will be closed.

**Comment by asteriskteam** [comment]
This PR has been closed because it has been in "Changes Requested" or "submitter-action-required" state for more than 42 days.

**Pull request closed without merging**